src/sonic_ax_impl/mibs/vendor/cisco/ciscoPfcExtMIB.py

```diff
@@ -0,0 +1,263 @@
+from enum import unique, Enum
+from bisect import bisect_right
+
+from sonic_ax_impl import mibs
+from ax_interface import MIBMeta, ValueType, MIBUpdater, MIBEntry, SubtreeMIBEntry
+from ax_interface.encodings import ObjectIdentifier
+
+class PfcUpdater(MIBUpdater):
+
+    def reinit_data(self):
+        """
+        Subclass update interface information
+        """
+        self.if_name_map, \
+        self.if_alias_map, \
+        self.if_id_map, \
+        self.oid_sai_map, \
+        self.oid_name_map = mibs.init_sync_d_interface_tables(self.db_conn)
+
+    def update_data(self):
+        """
+        Update redis (caches config)
+        Pulls the table references for each interface.
+        """
+        self.reinit_data()
+
+        self.if_counters = \
+            {sai_id: self.db_conn.get_all(mibs.COUNTERS_DB, mibs.counter_table(sai_id), blocking=True)
+             for sai_id in self.if_id_map}
+
+        self.lag_name_if_name_map, \
+        self.if_name_lag_name_map, \
+        self.oid_lag_name_map = mibs.init_sync_d_lag_tables(self.db_conn)
+
+        self.if_range = sorted(list(self.oid_sai_map.keys()) + list(self.oid_lag_name_map.keys()))
+        self.if_range = [(i,) for i in self.if_range]
+
+    def __init__(self):
+        super().__init__()
+        self.db_conn = mibs.init_db()
+        self.reinit_data()
+
+        self.lag_name_if_name_map = {}
+        self.if_name_lag_name_map = {}
+        self.oid_lag_name_map = {}
+
+        # cache of interface counters
+        self.if_counters = {}
+        self.if_range = []
+        # call our update method once to "seed" data before the "Agent" starts accepting requests.
+        self.update_data()
```

Comment:
>update_data [](start = 13, length = 11)

Is it really needed? What will happen if there is no 'seed' data?

Reply:
Removed double reinit_data()